airflow/providers/amazon/aws/hooks/glue.py

```diff
@@ -162,8 +165,11 @@ def initialize_job(
         run_kwargs = run_kwargs or {}
 
         try:
-            job_name = self.create_or_update_glue_job()
-            return self.get_conn().start_job_run(JobName=job_name, Arguments=script_arguments, **run_kwargs)
+            if self.update_config:
+                self.create_or_update_glue_job()
+            return self.get_conn().start_job_run(
```

Comment:
Thanks for the fast turnaround!

I don't think this quite matches the behaviour before your config update changes. Previously the code would check for an existing job with that name and then start it OR it would create the job if it didn't find an existing one, and then start that one. In this PR it seems if `update_config` is False (the default) we go straight to starting the job without first creating it if it doesn't exit (but I may be missing something).
I think you want something like:
```Python
try:
    if self.update_config:
        job_name = self.create_or_update_glue_job()
    else:
        job_name = self.get_or_create_glue_job()

    return glue_client.start_job_run(JobName=job_name, Arguments=script_arguments, **run_kwargs)

except Exception as general_error:
    self.log.error("Failed to run aws glue job, error: %s", general_error)
    raise
```

Reply:
Hey @o-nikolas, oh nice catch indeed. this is exactly what the code was doing before, I am gonna change it